### PR TITLE
[SYCL] Test for fix of linked alloca's deps

### PR DESCRIPTION
**Pull request by s-kanaev**
The tests provided here are for this case: https://github.com/intel/llvm/blob/sycl/sycl/source/detail/scheduler/graph_builder.cpp#L611.

The fix was introduced with intel/llvm#1471.

Use case fixed by the fix. Imagine us having two device queues (`Q1` and `Q2`) and a single buffer `B`. We initialize the buffer with host accessor. Also lets have two kernels: kernel `K1` executed at `Q1`; and kernel `K2` which executes at `Q2`. `K1` writes to `B`, `K2` reads from `B`. After submitting `K1` to `Q1` there will also be an `AllocaCommand` `A1` which allocates buffer on device. `K1` depends on `A1` via memory object `B`. After submitting `K2` to `Q2` there will be another `AllocaCommand` `A2` (for queue `Q2` and its device). `K2` will depend on `A2`. `A2`, however, should depend on both `A1` and `K1`. `A2->K1` dependency eliminates data race.

### CODE REVIEW

**Comment by s-kanaev** [comment]
> Please, add a test.


Done

File: sycl/test/scheduler/LinkedAlloca.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %sycl_source_dir %s -o %t.out
+// RUN: env SYCL_PRINT_EXECUTION_GRAPH=always %t.out
+// RUN: cat graph_7after_addHostAccessor.dot
+// RUN: cat graph_7after_addHostAccessor.dot | FileCheck %s
+
```
Comment (romanovvlad) [inline sycl/test/scheduler/LinkedAlloca.cpp:5]: Can we have "unit" test instead? Like this one unittests/scheduler/BlockedCommands.cpp ?

Comment (s-kanaev) [reply sycl/test/scheduler/LinkedAlloca.cpp:5]: Done.

File: sycl/unittests/scheduler/LinkedAllocaDependencies.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+class MemObjMock : public cl::sycl::detail::SYCLMemObjI {
+public:
+  using ContextImplPtr = std::shared_ptr<cl::sycl::detail::context_impl>;
+
+  MemObjMock(const std::shared_ptr<cl::sycl::detail::MemObjRecord> &Record)
+      : SYCLMemObjI() {
+    MRecord = Record;
+  }
+
+  ~MemObjMock(){};
```
Comment (alexbatashev) [inline sycl/unittests/scheduler/LinkedAllocaDependencies.cpp:23]:
```suggestion
  ~MemObjMock() = default;
```

Comment (s-kanaev) [reply sycl/unittests/scheduler/LinkedAllocaDependencies.cpp]: Done.

File: sycl/test/scheduler/LinkedAlloca.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      Acc[Idx] = -1;
+  }
+
+  // 2. submit kernel
+  Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        GeneratorAcc(Buf, CGH);
+
+    auto GeneratorKernel = [GeneratorAcc]() {
```
Comment (romanovvlad) [inline sycl/test/scheduler/LinkedAlloca.cpp:51]: Can we have a googletest/unittest instead and just check if a linked alloca is in the dependency list of a new alloca? Examples:
https://github.com/intel/llvm/blob/sycl/sycl/unittests/scheduler/BlockedCommands.cpp
https://github.com/intel/llvm/blob/sycl/sycl/unittests/scheduler/FinishedCmdCleanup.cpp

Comment (s-kanaev) [reply sycl/test/scheduler/LinkedAlloca.cpp:51]: There is a unit-test. It's added within this PR. See https://github.com/intel/llvm/pull/1470/files#diff-d96769fc46c97d667b6eb4625b44970dR1.

Comment (romanovvlad) [reply sycl/test/scheduler/LinkedAlloca.cpp:51]: Great! I didn't get to it.
So, why do we need LinkedAlloca.cpp test? Can we have LinkedAllocaDependencies.cpp  ?

Comment (romanovvlad) [reply sycl/test/scheduler/LinkedAlloca.cpp:51]: Repeating question to make it clear:
Why do we need LinkedAlloca.cpp test? Isn't LinkedAllocaDependencies.cpp enough?

Comment (s-kanaev) [reply sycl/test/scheduler/LinkedAlloca.cpp:51]: `sycl/test/scheduler/LinkedAlloca.cpp` isn't needed as it only looks at the graph and is not really checking data structures.

**Comment by s-kanaev** [comment]
@romanovvlad, please, review

**Pull request merged**